src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,32 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object rectangle) {
+        if (rectangle == this) {
+            return true;
+        }
+        if (rectangle == null) {
+            return false;
+        }
+        if (rectangle.getClass().equals(Rectangle.class)) {
+            Rectangle currentRectangle = (Rectangle) rectangle;
+            return ((this.width == currentRectangle.width || (this.width != null
+                    && this.width.equals(currentRectangle.width)))
+                    && (this.length == currentRectangle.length
+                    || (this.length != null && this.length.equals(currentRectangle.length)))
+                    && (this.color == currentRectangle.color || (this.color != null
+                    && this.color.equals(currentRectangle.color))));
```

Comment:
let's not use `this` keyword - it is redundant

Reply:
Fixed